etl_pipeline/utils/calculation_lineage.py

```
@@ -218,6 +218,37 @@ def add_aggregate_step(self, operation: str, fields: List[str],
             self.add_step(operation, field=",".join(fields), value=value, 
                          description=description, fields=fields)
     
+    def add_file_contribution(self, file_name: str, field_name: str, raw_value: Union[float, int, Decimal], 
+                             normalized_value: Union[float, int, Decimal], description: str = None) -> None:
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)